HIP-1.md

```diff
@@ -0,0 +1,146 @@
+    // Used by relayers to determine which off-chain metadata to inject, and
+    // how to format it.
+    // The first byte must contain the module "type", which informs the relayer
+    // how to interpret the remaining bytes.
+    function moduleMetadata() external view returns (bytes memory);
+
+    // Called by the Inbox to determine whether the provided root
+    // is valid to verify proofs against.
+    function accept(
+        bytes32 _root,
+        bytes calldata _message,
+        bytes calldata _metadata
+    ) external returns (bool);
+}
+
+interface IInbox {
+
+    // Called by relayers to deliver messages to the recipient.
+    // Verifies the merkle proof of `_message` against `_root` and
+    // calls `recipient.interchainSecurityModule().accept()` before
+    // calling `recipient.handle()`.
+    function process(
+        bytes32 _root,
+        bytes32[32] calldata _proof,
+        bytes calldata _message,
+        bytes calldata _metadata
+    ) external;
+}
+```
+
+### **Rationale**
+
+Sovereign consensus allows for a modular approach to interchain security.
+This allows the core protocol to be forward compatible, ensuring that Hyperlane can support new security models as they are developed.
+Furthermore, it allows applications to select, configure, or invent security models that offer the most appropriate tradeoffs.
+
+There were many ways to build this feature.
+We discuss a few of the design decisions made below.
+
+#### Location of merkle proof verification
+
+The ability for relayers to inject arbitrary metadata into the `ISM` presents a natural opportunity to move the merkle proof out of the `Inbox.process()` interface and into the `ISM`.
+In this model, the `Inbox` would be responsible only for replay protection.
+
+```
+interface IInterchainSecurityModule {
+    function moduleMetadata() external view returns (bytes memory);
+
+    function accept(
+        bytes32 _root,
+        bytes32[32] calldata _proof,
+        bytes calldata _message,
+        bytes calldata _metadata
+    ) external returns (bool);
+}
+
+interface IInbox {
+    function process(bytes calldata _message) external;
+}
+```
+
+This approach has the advantage of being even more modular than the proposed solution, as the `Inbox` is unopinionated about how messages are sent.
+This allows applications to create `ISMs` that do not incorporate the incremental merkle tree.
+As a simple example, one could build an `ISM` that is a multisig on individual messages, as opposed to a merkle root.
+
+The drawback is that those `ISMs` which wish to incorporate the incremental merkle tree into their security model must do that explicitly.
+We imagine most applications to prefer the censorship resistance and economies of scale that using the incremental merkle tree allows for.
+Thus, we propose leaving proof verification in the `Inbox` so that all `ISM` developers get this "for free".
+
+#### Removal of `_index` from Inbox.process()
```

Comment:
realizing this may also be referencing not passing the `_index` of a checkpoint into `ISM.accept()`. Passing it in as metadata is a bit more confusing, but I guess only the most advanced users would ever have to worry about implementing an ISM so I think that's reasonable.